ethereum/wait_for_event_task.go

```diff
@@ -0,0 +1,192 @@
+)
+
+const (
+	// ETHWaitForEvent is the name of the task
+	ETHWaitForEvent = "EthWaitForEventTask"
+
+	// WaitForEventFromBlock is the key for the from block
+	WaitForEventFromBlock = "WaitForEventFromBlock"
+
+	// WaitForEventAddress is the key for the address
+	WaitForEventAddress = "WaitForEventAddress"
+
+	// WaitForEventNameSignature is the key for the name and signature of the event
+	WaitForEventNameSignature = "WaitForEventNameSignature"
+
+	// WaitForEventTopic is the key for the topic
+	WaitForEventTopic = "WaitForEventTopic"
+)
+
+// WaitForEventTask holds from block, addresses of the contract to listen events from, topics of the event
+// the task assumes to block to be the latest
+type WaitForEventTask struct {
+	jobsv1.BaseTask
+	accountID             identity.DID
+	query                 ethereum.FilterQuery
+	filterLogsFunc        func(ctx context.Context, query ethereum.FilterQuery) ([]types.Log, error)
+	ethContextInitializer func(d time.Duration) (ctx context.Context, cancelFunc context.CancelFunc)
+}
+
+// Copy copies the state of the task into a new task
+func (t *WaitForEventTask) Copy() (gocelery.CeleryTask, error) {
+	return &WaitForEventTask{
+		BaseTask:              jobsv1.BaseTask{JobManager: t.JobManager},
+		filterLogsFunc:        t.filterLogsFunc,
+		ethContextInitializer: t.ethContextInitializer,
+	}, nil
+}
+
+// ParseKwargs parses the kwargs into a query.
+func (t *WaitForEventTask) ParseKwargs(kwargs map[string]interface{}) error {
+	err := t.ParseJobID(t.TaskTypeName(), kwargs)
+	if err != nil {
+		return err
+	}
+
+	accountID, ok := kwargs[TransactionAccountParam].(string)
+	if !ok {
+		return errors.NewTypedError(ErrEthTransaction, errors.New("missing account ID"))
+	}
+
+	fb, ok := kwargs[WaitForEventFromBlock].(string)
+	if !ok {
+		return errors.New("failed to decode from block: %v", kwargs[WaitForEventFromBlock])
+	}
+
+	address, ok := kwargs[WaitForEventAddress].(string)
+	if !ok {
+		return errors.New("failed to decode address: %v", kwargs[WaitForEventAddress])
+	}
+
+	eventSignature, ok := kwargs[WaitForEventNameSignature].(string)
+	if !ok {
+		return errors.New("failed to decode event name signature: %v", kwargs[WaitForEventNameSignature])
+	}
+
+	topic, ok := kwargs[WaitForEventTopic].(string)
+	if !ok {
+		return errors.New("failed to decode topic: %v", kwargs[WaitForEventTopic])
+	}
+
+	ehash := common.BytesToHash(crypto.Keccak256([]byte(eventSignature)))
+	t.accountID = identity.DID(common.HexToAddress(accountID))
+	t.query = ethereum.FilterQuery{
+		BlockHash: nil,
+		FromBlock: hexutil.MustDecodeBig(fb),
+		ToBlock:   nil,
+		Addresses: []common.Address{common.HexToAddress(address)},
+		Topics: [][]common.Hash{
+			{ehash},
+			{common.HexToHash(topic)},
+		},
+	}
+
+	return nil
+}
+
+// RunTask runs the task of fetching the logs.
+func (t *WaitForEventTask) RunTask() (res interface{}, err error) {
+	var jobValue *jobs.JobValue
+	ctx, cancelFunc := t.ethContextInitializer(5 * time.Second)
```

Comment:
Can we use `contextReadWaitTimeout` value from config instead or if it is still to high, create another constant with the appropriate value.